snowpack/src/scan-imports.ts

```diff
@@ -248,17 +251,21 @@ export async function scanImports(
   }
 
   // Scan every matched JS file for web dependency imports
-  const loadedFiles: (SnowpackSourceFile | null)[] = await Promise.all(
-    includeFiles.map(async (filePath) => {
+  const loadFileQueue = new PQueue({concurrency: CONCURRENT_WORKERS});
+  const getLoadedFiles = async (filePath: string): Promise<SnowpackSourceFile | null> =>
+    loadFileQueue.add(async () => {
       const {baseExt, expandedExt} = getExt(filePath);
       return {
         baseExt,
         expandedExt,
         locOnDisk: filePath,
         contents: await readFile(filePath),
       };
-    }),
+    });
+  const loadedFiles: (SnowpackSourceFile | null)[] = await Promise.all(
+    includeFiles.map(getLoadedFiles),
   );
+  await loadFileQueue.onIdle();
```

Comment:
If we've gotten all the results above, then we've gotten everything we need and don't need to wait for `onIdle()`

Reply:
Understood. It's the first time I've used this library. Thanks for the clarification.